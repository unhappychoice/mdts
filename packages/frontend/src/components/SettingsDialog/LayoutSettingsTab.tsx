import {
  Box,
  ToggleButton,
  ToggleButtonGroup,
  Typography
} from '@mui/material';
import { useTheme } from '@mui/material/styles';
import React, { useCallback } from 'react';

interface LayoutSettingsTabProps {
  contentMode: 'full' | 'compact';
  handleToggleContentMode: (mode: 'full' | 'compact') => void;
}

const LayoutSettingsTab: React.FC<LayoutSettingsTabProps> = ({ contentMode, handleToggleContentMode }) => {
  const theme = useTheme();

  const handleChange = useCallback((_, newMode) => {
    if (newMode)
      handleToggleContentMode(newMode);
  } , [handleToggleContentMode]);

  return (
    <Box sx={{ p: 2 }}>
      <Typography variant="subtitle2" sx={{ mb: 2 }}>Content Width</Typography>
      <ToggleButtonGroup
        value={contentMode}
        exclusive
        onChange={handleChange}
        aria-label="content width"
        fullWidth
        size="small"
        sx={{ mb: 2 }}
      >
        <ToggleButton
          value="full"
          aria-label="full"
          sx={{
            flexGrow: 1,
            '&.Mui-selected': {
              backgroundColor: theme.palette.primary.main,
              color: theme.palette.primary.contrastText,
              '&:hover': {
                backgroundColor: theme.palette.primary.dark,
              },
            },
          }}
        >
          Full
        </ToggleButton>
        <ToggleButton
          value="compact"
          aria-label="compact"
          sx={{
            flexGrow: 1,
            '&.Mui-selected': {
              backgroundColor: theme.palette.primary.main,
              color: theme.palette.primary.contrastText,
              '&:hover': {
                backgroundColor: theme.palette.primary.dark,
              },
            },
          }}
        >
          Compact
        </ToggleButton>
      </ToggleButtonGroup>
    </Box>
  );
};

export default LayoutSettingsTab;
