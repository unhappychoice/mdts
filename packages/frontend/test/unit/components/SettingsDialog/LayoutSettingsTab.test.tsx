import { render } from '@testing-library/react';
import React from 'react';
import LayoutSettingsTab from '../../../../src/components/SettingsDialog/LayoutSettingsTab';

describe('LayoutSettingsTab', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<LayoutSettingsTab />);
    expect(asFragment()).toMatchSnapshot();
  });
});
