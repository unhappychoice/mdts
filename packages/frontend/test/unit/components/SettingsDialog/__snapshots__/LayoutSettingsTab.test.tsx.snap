// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LayoutSettingsTab should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-19midj6"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-n2une3-MuiTypography-root"
    >
      Content Width
    </h6>
    <div
      aria-label="content width"
      class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal MuiToggleButtonGroup-fullWidth css-1jjbo4h-MuiToggleButtonGroup-root"
      role="group"
    >
      <button
        aria-label="full"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-zhjb61-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="full"
      >
        Full
      </button>
      <button
        aria-label="compact"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-zhjb61-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="compact"
      >
        Compact
      </button>
    </div>
  </div>
</DocumentFragment>
`;
